hooks/400-trim-pkcs-11.chroot

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh -ex
+
+mkdir -p /usr/share/p11-kit/modules
+echo "Creating p11-kit-client module configuration"
+echo "module: p11-kit-client.so" > \
+  /usr/share/p11-kit/modules/p11-kit-client.module
+
+# we should keep only following
+#   /usr/bin/p11-kit
+#   /usr/bin/p11tool
+#   /usr/bin/pkcs11-tool
+#   /usr/libexec
+#   /usr/lib/*/engines-*/*.so*
+#   /usr/lib/*/libopts.so*
+#   /usr/lib/*/libopensc.so*
+#   /usr/lib/*/libp11.so*
+#   /usr/lib/*/p11-kit-proxy.so*
+#   /usr/lib/*/pkcs11/*.so
+#   /usr/share/p11-kit
+
+echo "Cleaning excess pkcs#11 library"
+dpkg -r --force-depends libgnutls-dane0 libunbound8 libevent-2.1-7
+
+rm -rf \
+    /etc/opensc \
+    /usr/bin/cardos-tool \
+    /usr/bin/certtool \
+    /usr/bin/cryptoflex-tool \
+    /usr/bin/danetool \
+    /usr/bin/dnie-tool \
+    /usr/bin/egk-tool \
+    /usr/bin/eidenv \
+    /usr/bin/gids-tool \
+    /usr/bin/gnutls-cli \
+    /usr/bin/gnutls-cli-debug \
+    /usr/bin/gnutls-serv \
+    /usr/bin/goid-tool \
+    /usr/bin/iasecc-tool \
+    /usr/bin/netkey-tool \
+    /usr/bin/npa-tool \
+    /usr/bin/ocsptool \
+    /usr/bin/openpgp-tool \
+    /usr/bin/opensc-asn1 \
+    /usr/bin/opensc-explorer \
+    /usr/bin/opensc-notify \
+    /usr/bin/opensc-tool \
+    /usr/bin/piv-tool \
+    /usr/bin/pkcs11-register \
+    /usr/bin/pkcs15-crypt \
+    /usr/bin/pkcs15-init \
+    /usr/bin/pkcs15-tool \
+    /usr/bin/psktool \
+    /usr/bin/sc-hsm-tool \
+    /usr/bin/srptool \
+    /usr/bin/trust \
+    /usr/bin/westcos-tool \
+    /usr/lib/*/engines-1.1/pkcs11.la \
+    /usr/lib/*/libsmm-local.* \
+    /usr/lib/*/onepin-opensc-pkcs11.* \
+    /usr/lib/*/opensc-pkcs11.so \
+    /usr/lib/*/pkcs11/onepin-opensc-pkcs11.so \
+    /usr/lib/*/pkcs11/opensc-pkcs11.so \
+    /usr/lib/*/pkcs11-spy.* \
```

Comment:
Also remove the dead symlink `/usr/lib/*/pkcs11/pkcs11-spy.*`

Reply:
@valentindavid
 `/usr/share/doc/opensc/README.md` is link to `/usr/share/doc/opensc-pkcs11/README.md` which comes from `opensc-pkcs11` which we do not include. So I will trim that dead link as well
thank you for noticing those dead symbolic links!
yes core-base PR is also opened here: https://github.com/snapcore/core-base/pull/40

Reply:
@valentindavid as for `pkcs11` interface. I do not think this is needed, not for `p11-kit-client.so`, it's located at `/usr/lib/*/pkcs11/p11-kit-client.so`
Under `/usr/share/p11-kit/modules` is `opensc-pkcs11.module` which we do not intent to use from base snap. This would be typically included in gadget snap...